tests/ops/functions/test_generator.py

```diff
@@ -378,44 +394,62 @@ class TestHamiltonianReturn:
     """Tests for format="hamiltonian". This format always returns the generator
     as a Hamiltonian."""
 
+    @pytest.mark.usefixtures("use_legacy_opmath")
     def test_observable_no_coeff(self):
         """Test a generator that returns an observable with no coefficient is correct"""
         gen = qml.generator(qml.PhaseShift, format="hamiltonian")(0.5, wires=0)
         assert gen.name == "Hamiltonian"
         assert gen.compare(1.0 * qml.PhaseShift(0.5, wires=0).generator())
 
+    def test_observable_no_coeff_opmath(self):
+        """Test a generator that returns an observable with no coefficient is correct with opmath enabled"""
+        gen = qml.generator(qml.PhaseShift, format="hamiltonian")(0.5, wires=0)
+        assert gen.name == "LinearCombination"
+        gen_compare = qml.PhaseShift(0.5, wires=0).generator()
+        assert gen.ops[0] == gen_compare
+
+    # Also works with opmath disabled by replacing
+    # assert gen.name == "LinearCombination"   with   assert gen.name == "Hamiltonian"
     def test_observable(self):
         """Test a generator that returns a single observable is correct"""
         gen = qml.generator(ObservableOp, format="hamiltonian")(0.5, wires=0)

```

Comment:
```suggestion
    @pytest.mark.usefixtures("use_legacy_and_new_opmath")
    def test_observable(self):
        """Test a generator that returns a single observable is correct"""
        gen = qml.generator(ObservableOp, format="hamiltonian")(0.5, wires=0)
        if qml.operation.active_new_opmath():
            assert gen.name == "LinearCombination"
        else:
            assert gen.name == "Hamiltonian"

```

Optional, and honestly probably overkill/unnecessary